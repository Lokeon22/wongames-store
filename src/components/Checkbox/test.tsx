import { screen, waitFor } from "@testing-library/react"
import { renderWithTheme } from "../../utils/tests/helpers"

import userEvent from "@testing-library/user-event"
import Checkbox from "."

describe("<Checkbox />", () => {
  it("should render with label", () => {
    const { container } = renderWithTheme(
      <Checkbox label="checkbox label" labelFor="check" />
    )

    expect(screen.getByRole("checkbox")).toBeInTheDocument()
    expect(screen.getByLabelText(/checkbox label/i)).toBeInTheDocument()
    expect(screen.getByText(/checkbox label/i)).toHaveAttribute("for", "check")

    expect(container.firstChild).toMatchSnapshot()
  })

  it("should render without label", () => {
    renderWithTheme(<Checkbox />)

    expect(screen.queryByLabelText("checkbox")).not.toBeInTheDocument()
  })

  it("should render with black label", () => {
    renderWithTheme(
      <Checkbox label="checkbox label" labelFor="check" labelColor="black" />
    )

    expect(screen.getByText(/checkbox label/i)).toHaveStyle({
      color: "#030517"
    })
  })

  it("should dispatch onCheck when label change", async () => {
    const onCheck = jest.fn()

    renderWithTheme(<Checkbox label="check box" onCheck={onCheck} isChecked />)

    userEvent.click(screen.getByRole("checkbox"))
    await waitFor(() => expect(onCheck).toHaveBeenCalledTimes(1))

    expect(onCheck).toHaveBeenCalledWith(false)
  })
})
