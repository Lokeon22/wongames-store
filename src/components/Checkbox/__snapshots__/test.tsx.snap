// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render with label 1`] = `
.c0 {
  display: flex;
  align-items: center;
  gap: 0.5rem;
}

.c1 {
  width: 1.9rem;
  height: 1.9rem;
  display: flex;
  align-items: center;
  justify-content: center;
  appearance: none;
  border-radius: 0.2rem;
  border: 0.2rem solid #2E2F42;
  position: relative;
  outline: none;
  transition: background border 0.1s ease-in-out;
}

.c1::before {
  content: "";
  width: 0.6rem;
  height: 0.9rem;
  border: 0.2rem solid #FAFAFA;
  border-top: 0;
  border-left: 0;
  transform: rotate(45deg);
  position: absolute;
  top: 0.1rem;
  opacity: 0;
  transition: 0.1s ease-in-out;
}

.c1:focus {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c1:checked {
  background-color: #F231A5;
  background: #F231A5;
}

.c1:checked::before {
  opacity: 1;
}

.c2 {
  cursor: pointer;
  color: #FAFAFA;
  line-height: 1;
}

<main
  class="c0"
>
  <input
    class="c1"
    id="check"
    type="checkbox"
  />
  <label
    class="c2"
    for="check"
    labelcolor="white"
  >
    checkbox label
  </label>
</main>
`;
